tsfile/src/main/java/org/apache/iotdb/tsfile/file/metadata/TsFileMetaData.java

```diff
@@ -247,9 +271,61 @@ public int serializeTo(OutputStream outputStream) throws IOException {
     byteLen += ReadWriteIOUtils.write(totalChunkNum, outputStream);
     byteLen += ReadWriteIOUtils.write(invalidChunkNum, outputStream);
 
+
     return byteLen;
   }
 
+  /**
+   * use the given outputStream to serialize bloom filter.
+   *
+   * @param outputStream -output stream to determine byte length
+   * @return -byte length
+   */
+  public int serializeBloomFilter(OutputStream outputStream, List<ChunkGroupMetaData> chunkGroupMetaDataList)
+      throws IOException {
+    int byteLen = 0;
+    BloomFilter filter = buildBloomFilter(chunkGroupMetaDataList);
+
+    byte[] bytes = filter.serialize();
+    byteLen += ReadWriteIOUtils.write(bytes.length, outputStream);
+    outputStream.write(bytes);
+    byteLen += bytes.length;
+    byteLen += ReadWriteIOUtils.write(filter.getSize(), outputStream);
+    byteLen += ReadWriteIOUtils.write(filter.getHashFunctionSize(), outputStream);
+    return byteLen;
+  }
+
+  /**
+   * get all path in this tsfile
+   *
+   * @return all path in set
+   */
+  private List<String> getAllPath(List<ChunkGroupMetaData> chunkGroupMetaDataList) {
+    List<String> res = new ArrayList<>();
+    for (ChunkGroupMetaData chunkGroupMetaData : chunkGroupMetaDataList) {
+      String deviceId = chunkGroupMetaData.getDeviceID();
+      for (ChunkMetaData chunkMetaData : chunkGroupMetaData.getChunkMetaDataList()) {
+        res.add(deviceId + PATH_SEPARATOR + chunkMetaData.getMeasurementUid());

```

Comment:
um, I just wonder don't we have somewhere to store these sentence in memory?

Reply:
How about add the concatenated path into the bloom filter just within the loop?
Since you don't have to get all paths before you add them into the bloom filter.

Reply:
> Are you going to store all time series for each file in memory? That will surely explode.

I have no idea, i just consider concating the path again and again is time consuming.. (That is why I want to convert path to id)

> How about add the concatenated path into the bloom filter just within the loop?
Yes, but he must know the number of total paths, I am not sure whether he can get it.
```
BloomFilter bloomFilter = BloomFilter
        .getEmptyBloomFilter(TSFileDescriptor.getInstance().getConfig().getBloomFilterErrorRate(),
            paths.size());
```

Reply:
Ok, before no more idea can be provided, I will do not block the PR by this issue.